impeller/display_list/dl_dispatcher.cc

```diff
@@ -994,9 +996,26 @@ void TextFrameDispatcher::save() {
 
 void TextFrameDispatcher::saveLayer(const DlRect& bounds,
                                     const flutter::SaveLayerOptions options,
-                                    const flutter::DlImageFilter* backdrop) {
+                                    const flutter::DlImageFilter* backdrop,
+                                    std::optional<int64_t> backdrop_id) {
   save();
 
+  if (backdrop != nullptr && backdrop_id.has_value()) {
+    const auto& existing = backdrop_keys_.find(backdrop_id.value());
+    if (existing == backdrop_keys_.end()) {
+      backdrop_keys_[backdrop_id.value()] =
```

Comment:
No need to do a lookup again, you already found it.  Get a non-const reference.

Reply:
It depends on the constness of the collection.  If you have a non-const collection, like you do here the iterator will have a non-const reference.

Reply:
oh, the problem is that I was taking a reference to the iterator itself, which can only be const. BUt I don't need an iterator reference.

Reply:
Yea, stop taking the reference to the iterator, just take the iterator.  It seem like this could be buggy since you are taking the reference to something that is an rvalue.
```
std::unordered_map<int64_t, BackdropData>::iterator existing =
        backdrop_data_.find(backdrop_id.value());
```

Reply:
Done